Cinebase/Use Cases/Movies/MovieDetail/TagSelectionControl.swift

```diff
@@ -0,0 +1,375 @@
+//
+//  TagSelectionControl.swift
+//  Cinebase
+//
+//  Created by Charles Moncada on 5/17/17.
+//  Copyright © 2017 Globant. All rights reserved.
+//
+
+
+import UIKit
+
+class TagsSelectionControl: UIControl {
+    
+    var labels: [String] = [""] {
+        didSet {
+            reloadViews()
+        }
+    }
+    
+    var tagHeight: CGFloat? {
+        didSet {
+            reloadViews()
+        }
+    }
+    
+    var textFont: UIFont = UIFont.systemFont(ofSize: 12) {
+        didSet {
+           reloadViews()
+        }
+    }
+    
+    var paddingX: CGFloat = 2 {
+        didSet {
+            for tagView in tagViews {
+                tagView.paddingX = paddingX
+            }
+            rearrangeViews()
+        }
+
+    }
+    var paddingY: CGFloat = 2
+    
+    var hasSquareTags: Bool = false {
+        didSet {
+            reloadViews()
+        }
+    }
+    
+    func reloadViews() {
+        for view in tagViews {
+            view.removeFromSuperview()
+        }
+        
+        tagViews = []
+        var maxWidth: CGFloat = 0
+        var height: CGFloat?
+        for label in labels {
+            var size = label.size(attributes: [NSFontAttributeName: textFont]) 
+            size.width += paddingX * 2 + cornerRadius * 2
+            if maxWidth < size.width {
+                maxWidth = size.width
+            }
+        }
+
+        if hasSquareTags && tagHeight == nil {
+            height = maxWidth
+        } else {
+            height = tagHeight
+        }
+        
+        for label in labels {
+            let view = TagView(tagName: label, height: height, square: hasSquareTags)
+            addSubview(view)
+            tagViews.append(view)
+            view.delegate = self
+        }
+        
+        layoutSubviews()
```

Comment:
Since the `reloadViews()` method is triggered by a lot of properties, this method will be called many times. In order to reduce the amount of computation required, we could call `setNeedsLayout()` instead so that the layout is performed as little as possible.